src/sql/workbench/services/notebook/browser/sql/sqlSessionManager.ts

```diff
@@ -133,6 +133,16 @@ export class SqlSessionManager implements nb.SessionManager {
 		}
 		return Promise.resolve();
 	}
+
+	shutdownAll(): Thenable<void> {
+		return Promise.all(SqlSessionManager._sessions.map(session => {
```

Comment:
Shouldn't you be clearing the session map as well when you do this? Or do we keep closed sessions around for other purposes?

Reply:
By session map do you mean the _sessions array? The session being shut down is removed from the array here:
https://github.com/microsoft/azuredatastudio/blob/6b4c98892c1b2f9e8d71de58838af67e649ac109/src/sql/workbench/services/notebook/browser/sql/sqlSessionManager.ts#L128